### DISCUSSION TITLE
Batch not expiring even after date has passed

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Lern/Community#139] [Issues] [answered]
Hello All,

I have created a course and added a batch to it.
The batch is not expiring even after the date has passed, it still shows `Edit Details`.
After the date has passed I should be able to see the batch as Expired and a button to create a new batch.
But I cannot see it happening.
Is there any job that is failing? How is the batch expired?

![course-details-batch-expiry](https://github.com/Sunbird-Lern/Community/assets/65657373/e82e0208-5425-40ae-9af2-39ac58988b46)
 
@coolbung @komalm @Renuka-S @vinod303 @paritshivani @gandham-santhosh

**Comment by AmiableAnil** [comment]
@Xitija please share the batch read API response.

**Reply by Xitija** [reply]
Sure. Here it is.
`
{
    "id": "api.course.batch.read",
    "ver": "v1",
    "ts": "2023-07-18 06:19:44:288+0000",
    "params": {
        "resmsgid": null,
        "msgid": "75b838a1-b69e-c18b-3594-fb8b829362c8",
        "err": null,
        "status": "success",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": {
            "identifier": "01384114526933811237",
            "createdFor": [
                "0130701891041689600"
            ],
            "certTemplates": {},
            "endDate": "2023-07-18",
            "description": "",
            "updatedDate": "2023-07-18 11:34:57:298+0530",
            "batchId": "01384114526933811237",
            "cert_templates": {},
            "createdDate": "2023-07-17 16:39:30:541+0530",
            "createdBy": "5d757783-a86a-40cd-a814-1b6a16d37cb6",
            "mentors": [
                "73396acb-b466-4e3e-9869-bb82c462e1f0"
            ],
            "name": "Batch test invite only 17th july",
            "id": "01384114526933811237",
            "enrollmentType": "invite-only",
            "courseId": "do_1138388300769525761143",
            "enrollmentEndDate": null,
            "collectionId": "do_1138388300769525761143",
            "startDate": "2023-07-17",
            "status": 1
        }
    }
}
`

**Reply by Jayaprakash8887** [reply, marked as answer]
@Xitija Pls share the API response with ED team and check if there is any hard coding in the UI behaviour. Kindly create a thread in Sunbird-ED
